modules/spring-config/src/main/resources/META-INF/mule.xsd

```diff
@@ -4456,7 +4456,7 @@
     <xsd:complexType name="setPropertyType">
         <xsd:complexContent>
             <xsd:extension base="abstractAddPropertyTransformerType">
-                <xsd:attribute name="propertyName" type="attributeType" use="required"/>
+                <xsd:attribute name="propertyName" type="nonEmptyAttributeType" use="required"/>
```

Comment:
if you do this you should be consistent and do it on all extensions of abstractAddPropertyTransformerType